source/MRViewer/MRSurfaceManipulationWidget.cpp

```diff
@@ -81,11 +86,17 @@ void SurfaceManipulationWidget::init( const std::shared_ptr<ObjectMesh>& objectM
     initConnections_();
     mousePressed_ = false;
     mousePos_ = { -1, -1 };
+    oldPoints_ = obj_->mesh()->points;
+    if ( firstSessionInit_ )
+    {
+        oldMesh_ = std::make_shared<Mesh>( obj_->mesh()->cloneRegion( obj_->mesh()->topology.getValidFaces() ) );
+        firstSessionInit_ = false;
+    }
 }
```

Comment:
instead of having `firstSessionInit_` check `!oldMesh_`

Reply:
fixed